### ISSUE TITLE
Add an "orchid:update" rake task

### ISSUE THREAD

**Issue by techgique** [CDRH/orchid#181]
Add a rake task that simplifies updating files in the main app from newer versions in Orchid.

For each of the following files / facilities:
- config initializer
- public config and example
- private config example
- locale files
- asset pipeline / webpacker config
- Gemfile

Perform the following steps:
- diff the contents and show as a preview to see if Orchid has a newer version
- prompt y/N on whether to copy the Orchid version into the main app
- if y, add a `.bak` extension onto the existing file and copy Orchid's version in
- print a message with helpful suggestions if re-customization necessary on some lines
- automate restoration of customizations if the file allows for a demarcation between Orchid-wide and app-specific lines

The Gemfile portion of this may be more complicated than the others. This may inform building that later than the other parts, splitting it off as a separate rake task, or dropping it. Discussion will be welcome :)

**Comment by jduss4** [comment]
Agreed that the Gemfile is tricky, because we're not just copy pasting from any particular Gemfile in Orchid (it's a gemspec) so we would be gsub_file's lines of the Rails app Gemfiles....I think I'd prefer splitting that off as either a later task or something that needs to be manually managed!

Otherwise, I think that the files and steps look good, although I have never done something like diff the contents of a file. I have a feeling that will be tricky because orchid might have a new version OR the app might just have a custom version, and so we might want to use the same approach the Rails update script has, which is to always just copy in the new version (after a prompt) and then figure that users will figure out if they like the old version better.

**Comment by techgique** [comment]
We could add a Gemfile by creating a new Rails app from scratch, adding Orchid, and copying that file into Orchid itself. That might simplify things a lot. And we'd only have to do that process every once in a good while.

**Comment by jduss4** [comment]
That's true

**Comment by jduss4** [comment]
I am having trouble getting any generated tasks to run on my machine, and I'm wondering if it's a bundler settings issue or something.  Would anybody else mind trying to create a task and seeing it if shows up for them?

**Comment by jduss4** [comment]
Okay, in the interests of just getting SOME traction on this, and in the interests of being able to use all the helpful generator methods, I am going to make a simple generator.  In terms of the bells and whistles of detecting file changes, copying them, etc, I think I am going to leave that for another day.  For now, I will just add in some basic functionality for us to be able to prompt the user about changes they need to make for migration steps.
